src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -70,7 +72,7 @@ nodeExtractText = T.strip . mconcat . map extractText . nodeFlatten
     nodeFlatten (Node _pos ty subs) = ty : concatMap nodeFlatten subs
 
 data IgnoreMode
-  = Link
+  = Link (Maybe PosInfo)
```

Comment:
Let's add some haddocks explaining that `PosInfo` here represents the position where the `ignore link` annotation was declared.